### PR TITLE
Js contribution

### PR DESCRIPTION
**Pull request by james-swash**
Adding App designer complex docs

### CODE REVIEW

**Review by jagribble — APPROVED** [review]

File: src/projects/apps/designer/background.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Background
+
+The Background properties are used to add background effects for components.
+
+From the properties of the component you can select:
+- **Upload an image**: From your local storage.
```
Comment (chrismilway) [inline src/projects/apps/designer/background.md:6]: shouldn't use `local storage`. `from your computer` would be a better fit.

File: src/projects/apps/designer/background.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- **Image Repeat**: Sets if/how a background image will be repeated. This property may have the following values:
+ 
+| Setting | Description |
+| --- | --- |
+| _Initial_ | Sets this property to its default value. |
+| _No Repeat_ | The background image is not repeated. The image will only be shown once. |
+| _Repeat_ | The background image is repeated both vertically and horizontally. |
+| _Repeat X_ | The background image is repeated only horizontally. |
+| _Repeat Y_ | The background image is repeated only vertically |
+| – _Revert_ | Lets you 'undo' styles applied to a component, going back to the parent component. |
```
Comment (chrismilway) [inline src/projects/apps/designer/background.md:19]: got an erroneous `-` here

File: src/projects/apps/designer/margin_border_padding.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+- **Margin** - Defines the area between the component's border and the components around it
+
+
+![Box Model](/src/assets/box_model.png)
+
+In web development, this is referred to as the "Box Model", you can learn more about this [here](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/The_box_model). 
+
+### Margin
+
+Let's explore **margins** first. Consider the component illustrated below, which has a margin of 10 pixels:
```
Comment (chrismilway) [inline src/projects/apps/designer/margin_border_padding.md:18]: lol

File: src/projects/apps/designer/margin_border_padding.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# Margin & Border & Padding
```
Comment (chrismilway) [inline src/projects/apps/designer/margin_border_padding.md:1]: While it makes sense for this to be a single file from the authoring perspective, it's not so good from the UI perspective. Basically where does the button go? Margin, padding and border are 3 different panels

File: src/projects/apps/designer/shadow.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Shadow
+
+The shadow property adds shadow effects around a component's frame. This can give the illusion of a component having depth or appearing "above" the layout the component sits within.
+
+From the properties of the component you can select:
+- **colour**: Specify the colour of the shadow
+- **X**: Horizontal offset value
+- **Y**: Vertical offset value
+- **Blur**: Add a Blur Effect to the Shadow
+- **Spread**: Set the Spread Radius of the Shadow
```
Comment (chrismilway) [inline src/projects/apps/designer/shadow.md:10]: I think spread could do with a bit more explanation

File: src/projects/apps/designer/size.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# Size
+
+This property is useful to control the size of the component. The **height** and **width** properties are used to set the height and width of a component. It sets the height/width of the area inside the padding, border, and margin of the component.
```
Comment (chrismilway) [inline src/projects/apps/designer/size.md:3]: It does not set the height _within_ the padding - everything within apps inherits `box-sizing: border-box` (unless they explicitly opt-out), meaning that specified heights include the padding.

File: src/projects/apps/designer/size.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+The **height** and **width** properties may have the following values:
+- **Automatic**: Using "auto" for width and height is when you want components to be flexible and responsive to different content sizes and screen sizes. It allows the browser to handle the layout dynamically based on the content's dimensions.
+- **Fluid**: Fluid is a percentage, typically of whatever its parent component is. This should always work with width, but with height, it will only work when the parent has a fixed height.
+- **Fixed**: Fixed is a hard-coded value based on the theme spacing units. A spacing unit is *usually* 4px
+
+  Only the layout components also have the value of _Full width_ (where the horizontal and vertical length ratio is the same).
```
Comment (chrismilway) [inline src/projects/apps/designer/size.md:10]: This doesn't really explain it at all. Better would be

```
- **Full width**: Set to the full width of the browser window. **Only available to layout components, and will only work for top-level components on a page**.
```

**Pull request merged**